task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -296,8 +296,22 @@ def run(ti: RuntimeTaskInstance, log: Logger):
         )
 
         # TODO: Run task failure callbacks here
-    except (AirflowTaskTimeout, AirflowException, AirflowTaskTerminated):
+    except AirflowTaskTerminated:
         ...
+    except (AirflowTaskTimeout, AirflowException):
+        # Couldn't load the task, don't know number of retries, guess
+        if not getattr(ti, "task", None):
+            # Let us set the task_retries to default = 0
+            msg = RetryTask(
+                end_date=datetime.now(tz=timezone.utc),
+                task_retries=0,
+            )
+        else:
+            msg = RetryTask(
+                end_date=datetime.now(tz=timezone.utc),
+                # is `or 0` needed?
+                task_retries=ti.task.retries or 0,
+            )
```

Comment:
As for the API and this PR, all we should care about is whether we should retry or not. The task ran, complained that it needs to retry, so we send a retry API call. The core logic of how retry works should be out of the scope of this PR.